### PR TITLE
Added Joke Of Day Plugin

### PR DESCRIPTION
**Pull request by moksh-shukla**
this plugin gives you joke of the day which will surely bring a smile on your face

### CODE REVIEW

**Comment by moksh-shukla** [comment]
@appi147 updated pull request with all the previous issues solved

**Comment by moksh-shukla** [comment]
@pnhofmann please check this

File: jarviscli/plugins/joke_of_day.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+from plugin import plugin, require
+import requests
+from colorama import Fore
+from plugins.animations import SpinnerThread
+
+
+@require(network=True)
+@plugin('jokeofday')
```
Comment (pnhofmann) [inline jarviscli/plugins/joke_of_day.py]: Would suggest we name this plugin just 'joke'. And rename current 'joke' plugin to 'joke chuck'.

File: jarviscli/plugins/joke_of_day.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            self.joke(jarvis, joke_fetch)
+
+    def get_joke(self, jarvis):
+        spinner = SpinnerThread('Fetching ', 0.15)
+        while True:
+            joke = jarvis.input(
+                "Select your choice \n 1: Get Joke Of Day \n 2: Exit\n", Fore.GREEN)
+            print()
+            if joke == '1':
+                url = "https://api.jokes.one/jod"
```
Comment (pnhofmann) [inline jarviscli/plugins/joke_of_day.py]: From https://jokes.one/terms#apiuse:

> You can reproduce the jokes returned from the API in your applications, but you need to **credit Jokes One in your terms or about page**.

File: jarviscli/plugins/joke_of_day.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        jarvis.say("Welcome To The Plugin Joke Of Day!", Fore.CYAN)
+        print()
+        joke_fetch = self.get_joke(jarvis)
+        if joke_fetch is not None:
+            self.joke(jarvis, joke_fetch)
+
+    def get_joke(self, jarvis):
+        spinner = SpinnerThread('Fetching ', 0.15)
+        while True:
+            joke = jarvis.input(
```
Comment (pnhofmann) [inline jarviscli/plugins/joke_of_day.py]: Don't actually see the use for this user input. If user enters 'jokeofday' - he probably want's the joke of the day :).

Or do you e.g. plan to add features like 'random joke'? If yes, you could also add two plugins: "joke" for e.g. random joke and 'joke day" for the joke of the day.

File: jarviscli/plugins/joke_of_day.py (hunk 2 above)
Comment (moksh-shukla) [reply jarviscli/plugins/joke_of_day.py]: attribution given

**Comment by moksh-shukla** [comment]
@pnhofmann chuck and joke of day have been merged under a single heading joke with two plugins joke daily and joke chuck

**Comment by moksh-shukla** [comment]
@pnhofmann

**Pull request merged**